bson/src/main/org/bson/codecs/pojo/ConventionAnnotationImpl.java

```diff
@@ -79,6 +79,9 @@ private void processPropertyAnnotations(final ClassModelBuilder<?> classModelBui
                     propertyModelBuilder.readName(bsonProperty.value());
                 }
                 propertyModelBuilder.discriminatorEnabled(bsonProperty.useDiscriminator());
+                if (propertyModelBuilder.getName().equals(classModelBuilder.getIdPropertyName())) {
+                    classModelBuilder.idPropertyName(null);
+                }
             } else if (annotation instanceof BsonId) {
                 classModelBuilder.idPropertyName(propertyModelBuilder.getName());
             } else if (annotation instanceof BsonNotId) {
```

Comment:
In trying to understand the latest patch, I realize that there might be a bug in the first patch.  Consider a class with the following fields:

    @BsonId
    private String foo;

    @BsonNotId
    private long id;

I think line 85 will cause the "foo" property to serialize as "foo" instead of "_id".  Line 82-83 seems like a similar bug.